### DISCUSSION TITLE
Streaming inference with microphone

### DISCUSSION THREAD

**Discussion by manuel3265** [NVIDIA-NeMo/NeMo#5013] [General]
Hello, I would like to know if there is any way to use the microphone for streaming inferences, but that the captured audio is a maximum of one second and that still generates the transcript.

https://github.com/NVIDIA/NeMo/blob/main/examples/asr/asr_streaming/speech_to_text_streaming_infer.py

I have seen this example but I see that they do it with audio files, is there any way to do it with the microphone?

**Comment by VahidooX** [comment]
It is recommended to use this script for models trained in streaming mode. For models trained regularly in offline mode, you may use buffered streaming. How to use microphone is dependent to your environment, but you may take a look at this tutorial on how to do it in notebooks:

https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Online_ASR_Microphone_Demo.ipynb
